docs/source/run/service.rst

```diff
@@ -0,0 +1,93 @@
+====================
+Running as a service
+====================
+
+Now that you can run the bot, you probably want it to run forever in the
+background of your system. While there are many ways to do so, there are a few
+steps and requirements that remain the same to properly run Sopel as a service.
+
+
+Requirements to run a service
+=============================
+
+Sopel needs the following:
+
+* a non-root user to run as
+* read/write access to its configuration file
+* a writable pid directory
+  (:attr:`~sopel.config.core_section.CoreSection.pid_dir`)
+* a writable logs directory
+  (:attr:`~sopel.config.core_section.CoreSection.logdir`)
+* a writable data directory
+  (:attr:`~sopel.config.core_section.CoreSection.homedir`)
+
+.. important::
+
+    Do **not** run Sopel as the **root** user. And avoid if possible to run
+    with a privileged user.
+
+
+Running with systemd
+====================
+
+`systemd`__ is a service manager used by Debian and most Debian-derived
+Linux distributions. Here is an example of a service unit file to run Sopel as
+a service, placed at ``/etc/systemd/system/sopel.service``:
+
+.. code-block:: systemd
```

Comment:
This should link to `contrib/sopel.service` and `contrib/sopel@.service`, not duplicate them.

Reply:
I think I'll remove them from contrib, that would be even better.

@dgw I feel like the contrib folder could be modified in another PR instead of this one, because @half-duplex is right about the issue of duplicating the same information.

Reply:
@Exirel If you're moving stuff from `contrib` into the docs, it's fair to delete the source in the same PR so as to keep the change atomic.

Reply:
Alright that's fair! Thanks. :+1:

Reply:
I disagree with removing it from contrib/.
- This results in having service files in two completely unrelated places to keep in sync
- The single-instance one is no longer easily available for packagers or scripts to use, which means more users and packagers will write their own, with varying quality

Reply:
@half-duplex Please suggest the optimal outcome (in your opinion), then, given the concerns:

- Maintaining service files in multiple locations (in docs _and_ `contrib/`) _will_ lead to desyncs when one is inevitably forgotten while updating the other.
- The documentation should have the relevant information wholly contained within it, and not require following a link.

I share your concern about packagers/scripts losing access to templates; pointed out the same in my review note on `contrib/sopel.cfg`, though that also suffers from the desync possibility vs. examples in the docs as mentioned above.

Reply:
To me, the least bad option by a good margin is to just link the files in `contrib/`. It's better to make the user click a link than to inevitably end up with something _incorrect_ in the docs.
Alternatively, does sphinx have a way to just embed the file?

Reply:
> The single-instance one is no longer easily available for packagers or scripts to use, which means more users and packagers will write their own, with varying quality

Why would we end up with something incorrect in the docs? Explain to me how is that going to happen if we have one and only one place for that information?

We don't maintain well the contrib directory, we overlooked it for most of the time. We put things in here that are mostly "do what you want with these, they are not part of our supported stuff, but sharing is cool so here it is". As a matter of fact, the single-instance unit was modified only 4 times in the last 8 years, including 2 changes unrelated to its quality: the version I rewrote is actually better now, following systemd's behavior and dedicated folders for logs, data, config, and run files (such as PID). From my perspective, it's *better* now in the doc than the version that was in contrib until now.

I'll move the multi-instance stuff into the doc, because I can get that it can be confusing to have two places for the same kind of information.